sdk/storage_blobs/src/blob/operations/get_properties.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{blob::Blob, prelude::*};
+use azure_core::{headers::*, prelude::*, RequestId};
+use chrono::{DateTime, Utc};
+
+#[derive(Debug, Clone)]
+pub struct GetPropertiesBuilder {
+    blob_client: BlobClient,
+    blob_versioning: Option<BlobVersioning>,
+    timeout: Option<Timeout>,
+    lease_id: Option<LeaseId>,
+    client_request_id: Option<ClientRequestId>,
+}
+
+impl GetPropertiesBuilder {
+    pub(crate) fn new(blob_client: BlobClient) -> Self {
+        Self {
+            blob_client,
+            blob_versioning: None,
+            timeout: None,
+            lease_id: None,
+            client_request_id: None,
+        }
+    }
+
+    setters! {
+        blob_versioning: BlobVersioning => Some(blob_versioning),
+        timeout: Timeout => Some(timeout),
+        lease_id: LeaseId => Some(lease_id),
+        client_request_id: ClientRequestId => Some(client_request_id),
+    }
+
+    pub fn into_future(self) -> Response {
+        Box::pin(async move {
+            let mut url = self.blob_client.url_with_segments(None)?;
+
+            self.timeout.append_to_url_query(&mut url);
+            self.blob_versioning.append_to_url_query(&mut url);
+
+            let mut request =
+                self.blob_client
+                    .prepare_request(url.as_str(), http::Method::HEAD, None)?;
+            request.add_optional_header(&self.lease_id);
+            request.add_optional_header(&self.client_request_id);
+
+            let response = self
+                .blob_client
+                .http_client()
+                .execute_request_check_status(&request)
+                .await?;
+
+            // TODO: Fix this
+            //let blob = Blob::from_headers(&blob_name, &container_name, snapshot_time, &headers)?;
```

Comment:
Can we get more context on this?

Reply:
While I used `git mv` to rename all of the builders to operations, then appended the results, git does not show these as being renamed and therefore keeping the previous history.

I don't have context on this, it's a holdover from the previous implementation.